app/src/components/comobox/ComboBox.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import {
+  Button,
+  ComboBox as AriaComboBox,
+  ComboBoxProps as AriaComboBoxProps,
+  FieldError,
+  Input,
+  Label,
+  ListBox,
+  ListBoxItem,
+  ListBoxItemProps,
+  Popover,
+  Text,
+  ValidationResult,
+} from "react-aria-components";
+
+import { classNames, Icon, Icons } from "@arizeai/components";
+
+import { comboBoxCSS, comboBoxItemCSS, comboBoxPopoverCSS } from "./styles";
+
+export interface ComboBoxProps<T extends object>
+  extends Omit<AriaComboBoxProps<T>, "children"> {
+  label?: string;
+  description?: string | null;
+  errorMessage?: string | ((validation: ValidationResult) => string);
+  children: React.ReactNode | ((item: T) => React.ReactNode);
+}
+
+export function ComboBox<T extends object>({
```

Comment:
Well, maybe this is the right approach after thinking about it a tad if we want to abstract way things like sizing. @Parker-Stafford - let's keep for now.

Reply:
sounds good, i was thinking and i think this makes sense cause it allows us to set the styles more reliably as well, it's hard to know how someone might compose a combobox and that could lead to missing styles if we base them off of an implementation that isn't guaranteed to be used